src/Microsoft.AspNetCore.Server.IISIntegration/NativeMethods.cs

```diff
@@ -37,85 +29,212 @@ public enum REQUEST_NOTIFICATION_STATUS
+        }
+
+        public static void HttpStopIncomingRequests()
+        {
+            http_stop_incoming_requests();
+        }
+
+        public static unsafe HttpApiTypes.HTTP_RESPONSE_V2* HttpGetRawResponse(IntPtr pInProcessHandler)
+        {
+            return http_get_raw_response(pInProcessHandler);
+        }
+
+        public static void HttpSetResponseStatusCode(IntPtr pInProcessHandler, ushort statusCode, string pszReason)
+        {
+            Validate(http_set_response_status_code(pInProcessHandler, statusCode, pszReason));
+        }
+
+        public static unsafe int HttpReadRequestBytes(IntPtr pInProcessHandler, byte* pvBuffer, int cbBuffer, out int dwBytesReceived, out bool fCompletionExpected)
+        {
+            return http_read_request_bytes(pInProcessHandler, pvBuffer, cbBuffer, out dwBytesReceived, out fCompletionExpected);
+        }
+
+        public static void HttpGetCompletionInfo(IntPtr pCompletionInfo, out int cbBytes, out int hr)
+        {
+            http_get_completion_info(pCompletionInfo, out cbBytes, out hr);
+        }
+
+        public static void HttpSetManagedContext(IntPtr pInProcessHandler, IntPtr pvManagedContext)
+        {
+            Validate(http_set_managed_context(pInProcessHandler, pvManagedContext));
+        }
+
+        public static IISConfigurationData HttpGetApplicationProperties()
+        {
+            var iisConfigurationData = new IISConfigurationData();
+            Validate(http_get_application_properties(ref iisConfigurationData));
+            return iisConfigurationData;
+        }
+
+        public static bool HttpTryGetServerVariable(IntPtr pInProcessHandler, string variableName, out string value)
+        {
+            return http_get_server_variable(pInProcessHandler, variableName, out value) == 0;
+        }
+
+        public static bool HttpShutdown()
+        {
+            return http_shutdown();
+        }
+
+        public static unsafe int HttpWebsocketsReadBytes(IntPtr pInProcessHandler, byte* pvBuffer, int cbBuffer, PFN_WEBSOCKET_ASYNC_COMPLETION pfnCompletionCallback, IntPtr pvCompletionContext, out int dwBytesReceived, out bool fCompletionExpected)
+        {
+            return http_websockets_read_bytes(pInProcessHandler, pvBuffer, cbBuffer, pfnCompletionCallback, pvCompletionContext, out dwBytesReceived, out fCompletionExpected);
+        }
+
+        public static unsafe int HttpWebsocketsWriteBytes(IntPtr pInProcessHandler, HttpApiTypes.HTTP_DATA_CHUNK* pDataChunks, int nChunks, PFN_WEBSOCKET_ASYNC_COMPLETION pfnCompletionCallback, IntPtr pvCompletionContext, out bool fCompletionExpected)
+        {
+            return http_websockets_write_bytes(pInProcessHandler, pDataChunks, nChunks, pfnCompletionCallback, pvCompletionContext, out fCompletionExpected);
+        }
+
+        public static void HttpEnableWebsockets(IntPtr pInProcessHandler)
+        {
+            Validate(http_enable_websockets(pInProcessHandler));
```

Comment:
if someone call this methods on Win7, I think there will an AV. the native method  in \src\RequestHandler\managedexports.cxx should do null check and condition check

Reply:
Can you please file a separate issue for that

Reply:
Don't think this is an issue, we will never call this method if websockets are disabled. We can check in native but it seems redundant.